### PR TITLE
build(deps)!: update to pydantic 2

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

### CODE REVIEW

**Review by tigarmo — APPROVED** [review]

File: craft_store/models/_common_list_releases_model.py [hunk 1]
```diff
@@ -41,8 +41,8 @@ class ChannelsModel(MarshableModel):
     :param track: the channel track.
     """
 
-    branch: str | None
-    fallback: str | None
+    branch: str | None = None
+    fallback: str | None = None
```
Comment (dariuszd21) [inline craft_store/models/_common_list_releases_model.py:45]: Shouldn't we put them at the end if they have default value to make it compliant with dataclasses?

File: pyproject.toml [hunk 2]
```diff
@@ -11,7 +11,7 @@ dependencies = [
     "requests>=2.27.0",
     "requests-toolbelt>=1.0.0",
     "macaroonbakery>=1.3.0,!=1.3.3",
-    "pydantic>=1.10,<2.0",
+    "pydantic~=2.0",
```
Comment (dariuszd21) [inline pyproject.toml:14]:
```suggestion
    "pydantic~=2.8",
```
As a precaution I'd suggest to use latest release as a accepted minimum just in case

**Review by dariuszd21 — APPROVED** [review]
Looks good to me!

File: craft_store/models/_common_list_releases_model.py (hunk 1 above)
Comment (lengau) [reply craft_store/models/_common_list_releases_model.py:45]: pydantic doesn't have the same rules as dataclasses. In pydantic 1, `str | None` was the same as `Optional[str]`, which meant it had a default value of `None` anyway, so this isn't actually changing the defaults.

**Pull request merged**